src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -1267,10 +1267,18 @@ private void VisitExpr(ParseNode node)
                 Opcode skipPastFunctionBody = AddOpcode(new OpcodeBranchJump());
                 string functionStartLabel = GetNextLabel(false);
                 
+                needImplicitReturn = true;
                 nextBraceIsFunction = true;
                 VisitNode(node.Nodes[0]); // the braces of the anonymous function and its contents get compiled in-line here.
                 nextBraceIsFunction = false;
-
+                if (needImplicitReturn)
+                    // needImplicitReturn is unconditionally true here, but it's being used anyway so we'll find this block
+                    // of code later when we search for "all the places using needImplicitReturn" and perform a refactor
+                    // of the logic for adding implicit returns.
```

Comment:
Yes.  The reason for the check "needImplicitReturn" is that my original plan was to detect the lack of a return statement and only do this when it's missing.  But then I realized it would take a lot more work than I'd originally anticipated to do that check properly (it has to walk the path of all the if/else conditions and loops and see if there's a case where it is missing even though it is present in other cases).  At the time I wanted to get functions working at all, so I hardcoded it to true for the time being, with the idea it could be revisited later on if we care enough to spend that effort.

My reason for still including it here even though I know it's hardcoded to true here too, is so that if we ever do decide to make this smarter in the future, this bit of code will stick out in a search as being obviously the same sort of thing and therefore get noticed as needing to be changed as well as the compiler's main function code.